### DISCUSSION TITLE
onBeforeSubmit?

### DISCUSSION THREAD

**Discussion by ghost** [airjp73/rvf#42] [Q&A] [answered]
Is there an event before submit?

I want do some action before submit on client side.

**Comment by airjp73** [comment, marked as answer]
The regular `onSubmit` callback runs before the submission occurs, but after validation. Are you looking to do something before validation?

And if you want to abort the submit, you can call `preventDefault` as normal.

```tsx
<ValidatedForm onSubmit={(validatedData, event) => event.preventDefault()}
```
